src/cript/api/api.py

```diff
@@ -511,22 +510,42 @@ def save(self, project: Project) -> None:
 
         Returns
         -------
-        None
+        A set of extra saved node UUIDs.
             Just sends a `POST` or `Patch` request to the API
         """
+        if known_uuid is None:
+            known_uuid = set()
 
-        project.validate()
-
+        node.validate()
         # saves all the local files to cloud storage right before saving the Project node
         # Ensure that all file nodes have uploaded there payload before actual save.
-        for file_node in project.find_children({"node": ["File"]}):
+        for file_node in node.find_children({"node": ["File"]}):
             file_node.ensure_uploaded(api=self)
 
-        response: Dict = requests.post(url=f"{self._host}/{project.node_type.lower()}", headers=self._http_headers, data=project.json).json()
+        node_known = len(self.search(type(node), SearchModes.UUID, str(node.uuid)).current_page_results) == 1
 
+        json_data = node.get_json(known_uuid=known_uuid).json
+        if node_known:
+            response: Dict = requests.patch(url=f"{self._host}/{node.node_type.lower()}", headers=self._http_headers, data=json_data).json()
+        else:
+            response: Dict = requests.post(url=f"{self._host}/{node.node_type.lower()}", headers=self._http_headers, data=json_data).json()
+
+        if response["code"] == 400 and response["error"].startswith("Bad uuid:"):
```

Comment:
I catch this within the `brute_force_save` function and handle it there. I'd suggest using that if possible

Reply:
we can, use a helper function to catch this.
But your `brute_force_save` function, works fundamentally different then my save.

Reply:
okie dokes we can get something working for now and come back and refactor it after